nestjs/src/discord-servers/discord-servers.controller.ts

```diff
@@ -20,14 +21,19 @@ export class DiscordServersController {
   }
 
   @Get()
+  @RequiredRoles([Role.Admin, CourseRole.Manager])
   @ApiOperation({ operationId: 'getDiscordServers' })
   @ApiOkResponse({ type: [DiscordServerDto] })
-  public async getAll() {
+  public async getAll(@Req() req: Request) {
+    const user = req.user as AuthUser;
+    const reducedInfo = !user.isAdmin;
```

Comment:
Please consider using `CurrentRequest` type instead of Express `Request` type, so you won't need typecast
```suggestion
  public async getAll(@Req() req: CurentRequest) {
    const isAdmin = Boolean(req.user.isAdmin);
```